src/components/Search.js

```diff
@@ -77,8 +79,12 @@ export const Search = () => {
 
   useEffect(() => {
     articleHistory.clear()
-    setNavigation(0)
-  }, [])
+    if (!consentGranted) {
+      showConsentPopup()
+    } else {
+      setNavigation(0)
+    }
+  }, [consentGranted])
```

Comment:
`consentGranted` is not a state, just a variable, should be fine just use it in the effect

Reply:
When I test on device with the `useEffect` below, the result I'm seeing is that the cursor/focus is not placed on search bar after consent. 

```
useEffect(() => {
    articleHistory.clear()
    if (!isConsentGranted()) {
      showConsentPopup()
    } else {
      setNavigation(0)
    }
  }, [])
```

With `consentGranted` in the dependency array I do see the focus on search bar after grating consent on device

Reply:
also instrumentation code works only when the dependency array has `consentGranted`

🤔 don't understand why the `consentGranted` is not either prop or state but works

Reply:
A bit puzzling to me as well actually, as I understand it `consentGranted` is acting as state since `isConsentGranted` reads from `localStorage` after `onAgree` handler runs in `Consent`